### ISSUE TITLE
Compatibility issue with fastify-autoload route prefixing 

### ISSUE THREAD

**Issue by alitnk** [elierotenberg/fastify-zod#34]
[fastify-autoload ](https://github.com/fastify/fastify-autoload) handles adding route prefix for the loaded plugins, depending but it doesn't work when it's used with this library.

For example:
```js
fastify.get('/login', ...)
```
gets prefixed but
```js
fastify.zod.get('/login', ...)
```
does not get prefixed.

**Comment by r0mankon** [comment]
Yeah and it doesn't even get the `prefix` defined in autoload. Absolute killer for me!

**Comment by nikeee** [comment]
I just tried played around by migrating my server to this and ancountered this issue. This basically makes it implossible for me to adopt fastify-zod (getting `FST_ERR_DUPLICATED_ROUTE` due to this).
